File: src/main/java/com/ourcompany/jpa/Person.java
package com.ourcompany.jpa;

import jakarta.persistence.Cacheable;
import jakarta.persistence.Column;
import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;

@Cacheable
@Entity
@DiscriminatorValue("PERS")
public class Person extends User {
    @Column(name = "surname")
    String surname;
}

File: src/main/kotlin/com/ourcompany/jpa/Person.kt
package com.ourcompany.jpa

import jakarta.persistence.Cacheable
import jakarta.persistence.Column
import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity

@Cacheable
@Entity
@DiscriminatorValue("PERS")
class Person : User() {
    @Column(name = "surname")
    var surname: String? = null
}
